### PR TITLE
Merge Release 4.10.1 into Master

### PR DESCRIPTION
**Pull request by zxu123**

### CODE REVIEW

**Comment by paulb777** [comment]
Should not be merged until release is public

**Comment by wilhuff** [comment]
Right: we're just working through the merge conflicts here so that we can release when ready. The fixes on 4.10.1 don't apply to master cleanly.

**Comment by zxu123** [comment]
Ack. Will wait for the release go public. (Actually travis got this already: CocoaPods could not find compatible versions for pod "Firebase/Core": In Podfile: Firebase/Core (= 4.10.1)) =D

File: Firestore/Source/Remote/FSTDatastore.mm [hunk 1]
```diff
@@ -319,7 +319,8 @@ - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
             [FSTDatastore
                 prepareHeadersForRPC:rpc
                           databaseID:&self.databaseInfo->database_id()
-                               token:(result.is_valid() ? result.token() : absl::string_view())];
+                               token:(result.user().is_authenticated() ? result.token()
+                                                                       : absl::string_view())];
```
Comment (mikelehen) [inline Firestore/Source/Remote/FSTDatastore.mm]: FYI- I think this needs to be clang-formatted. I noticed it in my original commit, but if you clang-format, it reindents this whole block causing a bigger diff, so I didn't bother.  But once you merge this into mater, perhaps we should immediately follow up with a clang-format fixup PR or something.

Comment (mikelehen) [reply Firestore/Source/Remote/FSTDatastore.mm]: Oh! Actually I see you have a "fix lint" commit in here, so maybe you already took care of the issue.  If so, ignore me.

**Pull request merged**